# Ex2 - Getting and Knowing your Data

### Step 1. Import the necessary libraries

In [ ]:
import pandas as pd

### Step 2. Import the dataset from this [address](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv).

In [ ]:
from urllib import request

URL = "https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv"

response = request.urlretrieve(URL, "Chipotle.tsv")

### Step 3. Assign it to a variable called chipo.

In [ ]:
chipo = pd.read_csv("/content/Chipotle.tsv", sep="\t")

### Step 4. See the first 10 entries

In [ ]:
chipo.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

### Step 5. What is the number of observations in the dataset?

In [ ]:
chipo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [ ]:
chipo.tail(1)

order_id  quantity           item_name  \
4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

4622 entries

### Step 6. What is the number of columns in the dataset?

In [ ]:
chipo.info()

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

5 columns

### Step 7. Print the name of all the columns.

In [ ]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

### Step 8. How is the dataset indexed?

this is a realational dataset
number indexed

### Step 9. Which was the most-ordered item?

In [ ]:
most_ordered_item = chipo.groupby("item_name")["quantity"].sum().idxmax()
print(most_ordered_item)

Chicken Bowl


### Step 10. For the most-ordered item, how many items were ordered?

In [ ]:
most_ordered_item, quantity_ordered = chipo.groupby("item_name")["quantity"].sum().idxmax(), chipo.groupby("item_name")["quantity"].sum().max()
print(most_ordered_item)
print(quantity_ordered)

Chicken Bowl
761


### Step 11. What was the most ordered item in the choice_description column?

In [ ]:
most_choice_item = chipo.groupby("choice_description")["quantity"].sum().idxmax()
print(most_choice_item)

[Diet Coke]


### Step 12. How many items were orderd in total?

In [ ]:
chipo.groupby("order_id")["quantity"].sum().idxmax()

1443

### Step 13. Turn the item price into a float

In [ ]:
chipo['item_price'] = df['item_price'].str.replace('$', '').astype(float)

<ipython-input-57-d944a789898f>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  chipo['item_price'] = df['item_price'].str.replace('$', '').astype(float)


#### Step 13.a. Check the item price type

In [ ]:
print(chipo['item_price'].dtype)

float64


#### Step 13.b. Create a lambda function and change the type of item price

#### Step 13.c. Check the item price type

In [ ]:
print("Data type of 'item_price' column:", chipo['item_price'].dtype)

Data type of 'item_price' column: float64


### Step 14. How much was the revenue for the period in the dataset?

In [ ]:
chipo['item_price'].sum()

34500.16

### Step 15. How many orders were made in the period?

In [ ]:
chipo['order_id'].nunique()

1834

### Step 16. What is the average revenue amount per order?

In [ ]:
chipo['item_price'].sum() / chipo['order_id'].nunique()

18.811428571428575

### Step 17. How many different items are sold?

In [ ]:
chipo['item_name'].nunique()

50